src/deploy/functions/services/auth.ts

```diff
@@ -82,13 +85,24 @@ export class AuthBlockingService implements Service {
           functionUri: endpoint.uri!,
         },
       };
-    } else {
+    } else if (endpoint.blockingTrigger.eventType === events.v1.BEFORE_SIGN_IN_EVENT) {
       newBlockingConfig.triggers = {
         ...newBlockingConfig.triggers,
         beforeSignIn: {
           functionUri: endpoint.uri!,
         },
       };
+    } else if (endpoint.blockingTrigger.eventType === events.v1.BEFORE_SEND_EMAIL_EVENT) {
+      newBlockingConfig.triggers = {
+        ...newBlockingConfig.triggers,
+        beforeSendEmail: {
+          functionUri: endpoint.uri!,
+        },
+      };
+    } else {
+      throw new FirebaseError(
+        `Received invalid blocking trigger event type ${endpoint.blockingTrigger.eventType}`
```

Comment:
quick question - seems like the error msg will be thrown to customers using firebase CLI, would new error codes and msg needs some sort of approval?

Reply:
The error message will be wrapped in an error object that will have a code related to a failed deployment.